### ISSUE TITLE
Openmpi error in MPI_Recv: MPI_ERR_TRUNCATE

### ISSUE THREAD

**Issue by freifrauvonbleifrei** [SGpp/DisCoTec#10]
I set up DisCoTec on a relatively new machine, but some of the tests (and also examples) are giving me this error here:

```
$ mpiexec.openmpi -n 9 ./test_distributedcombigrid_boost --run_test=ftolerance,loadmodel,rescheduling
Running 10 test cases...
Running 10 test cases...
Running 10 test cases...
Running 10 test cases...
Running 10 test cases...
Running 10 test cases...
Running 10 test cases...
Running 10 test cases...
Running 10 test cases...
[ipvs-epyc1:3260534] *** An error occurred in MPI_Recv
[ipvs-epyc1:3260534] *** reported by process [2024538113,8]
[ipvs-epyc1:3260534] *** on communicator MPI COMMUNICATOR 10 CREATE FROM 8
[ipvs-epyc1:3260534] *** MPI_ERR_TRUNCATE: message truncated
[ipvs-epyc1:3260534] *** MPI_ERRORS_ARE_FATAL (processes in this communicator will now abort,
[ipvs-epyc1:3260534] ***    and potentially your MPI job)
```
I can run individual tests and they will sometimes fail or not. 
I have made absolutely sure that DisCoTec and boost link to the system's OpenMPI version 4.0.3 (even installed an extra boost via spack).
@obersteiner have you seen this error before?

**Comment by obersteiner** [comment]
I am not aware that I had this error before. Can you locate in which MPI_Recv the error occurs?

**Comment by freifrauvonbleifrei** [comment]
This proves to be hard, as I dont have X11 on the system, so no `xterm -e gdb` ... will investigate further.

**Comment by freifrauvonbleifrei** [comment]
@hirschsn commented that it might be caused by send buffers in MPI_Isend going out of scope. I have verified this by replacing all `MPI_Isend`s by `MPI_Issend` -- which works fine. Not sure about the exact location though, we have some candidate spots that might be prone to this error.

**Comment by hirschsn** [comment]
Sry for interfering. I meant that as an example for "works sometimes and sometimes not". If you don't miscalculate the message sizes on receiver size, MPI_ERR_TRUNCATE could be stray messages not previously matched by a receive.

**Comment by freifrauvonbleifrei** [comment]
Sorry, correction: it does not work with `MPI_Issend` either -- I ran the wrong test. The error does not occur with mpich 3.3.1 though.

**Comment by freifrauvonbleifrei** [comment]
This change https://github.com/SGpp/DisCoTec/commit/148dc84e2b468dbeac71fa8c02159448388d3e53 makes the program hang up, which means that another message with tag 0 was overtaking this class-send message before. Now I just need to figure out what the other call is...

**Comment by freifrauvonbleifrei** [comment]
OK, I think this is due to some sends and receives without tag (0 / MPI_ANY_TAG) that were just magically going right before. at the current state (e6a5eeb214ac614cfad48209c52cc50763f32920), it is now the tests `ftolerance` and `rescheduling` that are failing.

**Comment by freifrauvonbleifrei** [comment]
This should be solved now (on master with commit cb101f3)
